NitroxPatcher/Patches/Dynamic/CreatureEgg_Hatch_Patch.cs

```diff
@@ -0,0 +1,27 @@
+using System.Reflection;
+using NitroxClient.Communication.Abstract;
+using NitroxClient.GameLogic;
+using NitroxModel.DataStructures;
+using NitroxModel.Helper;
+using NitroxModel.Packets;
+
+namespace NitroxPatcher.Patches.Dynamic;
+
+/// <summary>
+/// Syncs egg deletion when hatching.
+/// </summary>
+public sealed partial class CreatureEgg_Hatch_Patch : NitroxPatch, IDynamicPatch
+{
+    public static readonly MethodInfo TARGET_METHOD = Reflect.Method((CreatureEgg t) => t.Hatch());
```

Comment:
I'm assuming we don't need to deal with telling the other clients that it has hatched as this will be dealt with by the metadata processor/ async born patch below and we won't cause a race condition by sending the destroyed entity first?